docs/proposals/resource-catalog.md

```diff
@@ -0,0 +1,222 @@
+
+
+We can reshape the catalog resource to more easily express them, secure them, and claim them.
+
+# Goals
+ 1.  cluster-admin should control which catalog exist.
+ 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
+ 3.  project-admin should be able to delegate per-catalog exposure.
+ 4.  project-editor should be able to claim an entry from the catalog.
+ 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
+ 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
```

Comment:
So the project admin wants to delegate access to create certain things in
his namespace to a catalog claim controller?

On Mon, Apr 25, 2016 at 11:31 AM, David Eads <notifications@github.com>
wrote:

> In docs/proposals/resource-catalog.md
> <https://github.com/ncdc/kubernetes/pull/8#discussion_r60933929>:
>
> > @@ -0,0 +1,222 @@
> > +
> > +
> > +We can reshape the catalog resource to more easily express them, secure them, and claim them.
> > +
> > +# Goals
> > + 1.  cluster-admin should control which catalog exist.
> > + 2.  cluster-admin should be able/disable per-catalog exposure on a per-project granularity.
> > + 3.  project-admin should be able to delegate per-catalog exposure.
> > + 4.  project-editor should be able to claim an entry from the catalog.
> > + 5.  ResourceCatalogClaims should be sync-able, but not require syncing.
> > + 6.  project-admin should be able to control the shape of claim realization and prevent unwanted injection.
>
> Define unwanted injection
>
> Project admin may want to claim something from catalog, but want to make
> sure that it can't create serviceaccounts, secrets, or some other resource
> type in his namespace.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ncdc/kubernetes/pull/8/files/a2acd970c6316ce196c7bb3f24fb57e1bbef647e#r60933929>
>